src/solvers/flattening/bv_pointers.cpp

```diff
@@ -798,8 +798,8 @@ void bv_pointerst::do_postponed(
         if(size_expr.is_nil())
           continue;
 
-        if(to_integer(size_expr, object_size))
-          continue;
+        object_size =
+          typecast_exprt::conditional_cast(size_expr, postponed.expr.type());
```

Comment:
Could the code around here be refactored so that it is trivial that `object_size` will always be initialised when used below? I'd suggest:

````c++
if(expr.id() != ID_symbol)
  continue;

const size_expr = size_of_expr(type, ns);
if(size_expr.is_nil())
  continue;

const exprt object_size =
  typecast_exprt::conditional_cast(size_expr, postponed.expr.type());
````

Reply:
Done!